testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/TokenIntrospectionTest.java

```diff
@@ -322,7 +349,26 @@ public void testIntrospectAccessToken() throws Exception {
         AbstractOIDCScopeTest.assertScopes("openid email profile", rep.getScope());
     }
 
+    @Test
+    public void testIntrospectAccessTokenWithoutScope() throws Exception {
+        oauth.realm("realmWithoutScopes").clientId("no-scope").openid(false).doLogin("test-user@localhost", "password");
```

Comment:
Just a small detail, but is it possible to avoid creation of new realm, but instead just update existing "test" realm and possibly create some new client without scopes in it (and then cleanup this client afterwards etc)? Creation of new realm has some overhead and makes a test a bit longer to execute than necessary.

Reply:
I actually asked about this in the [Keycloak Slack channel](https://cloud-native.slack.com/archives/C056HC17KK9/p1696537271544279) and the answer was that creating a new realm is the simplest approach.

The problem is that if I apply this modification before importing the test realm it would apply to all tests and this would fail other tests (I tried this). So I would have to somehow conditionally apply this modification for only this test case or extract this one testcase into a new test class. Not so nice I think...

Alternatively, I could try to update the already imported realm as part of the test case using the adminClient - I have not explored this option yet. Is this what you mean?

Reply:
Yes, exactly. That is what I meant. The ideal is, that if some test needs something, it can be set at the beginning of the test method and then reverted at the end (to not affect other test methods, which rely on the "default" behaviour). We use this approach on many places. For instance this method: https://github.com/keycloak/keycloak/blob/22.0.4/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/LoginTest.java#L181-L202

This is usually sufficient and more effective than always creating new realm, which is a bit time consuming.